proposing additions for type annotation (#12076)
dotnet/docs: docs/fsharp/style-guide/formatting.md

type MyFunBad = int->int->string
```

### Surround function arguments with white space

When defining a function, use white space around each argument.

```fsharp
// OK
let myFun (a: decimal) b c = a + b + c

// Bad
let myFunBad (a:decimal)(b)c = a + b + c
```

> ### type annotation

Reviewer: ```suggestion
### Type annotations
```

> #### Right-pad function argument type annotation with white space

Reviewer: ```suggestion
#### Right-pad function argument type annotations
```

> When defining arguments with type annotation, use white space after the `:` symbol:

Reviewer: ```suggestion
When defining arguments with type annotations, use white space after the `:` symbol:
```

```fsharp
// OK
let complexFunction (a: int) (b: int) c = a + b + c

// Bad
let complexFunctionBad (a :int) (b :int) (c:int) = a + b + c
```

> #### Surround definition type annotations with white space

Reviewer: ```suggestion
#### Surround return type annotations with white space
```

> In a let binding with type annotation, use white space after the `:` symbol:

Reviewer: ```suggestion
In a let-bound function or value type annotation (return type in the case of a function), use white space before and after the `:` symbol:
```

```fsharp
// OK

> let expensiveToCompute : int = 0

Reviewer: ```suggestion
let expensiveToCompute : int = 0 // Type annotation for let-bound value
```

> let myFun (a: decimal) b c : decimal = a + b + c

Reviewer: ```suggestion
let myFun (a: decimal) b c : decimal = a + b + c // Type annotation for the return type of a function
```
